client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,287 @@
+import { $, $a, removeFromDOM, asArray, escapeID } from '../domhelpers.js';
+import { Widget } from './widget.js';
+
+class ScoreBoard extends Widget {
+  constructor(object, surface) {
+    super(object, surface);
+
+    this.addDefaults({
+      movable: true,
+      width: 300,
+      height: 200,
+      layer: -1,
+      typeClasses: 'widget scoreboard',
+      usePlayerColors: true,
+      playersInColumns: true,
+      rounds: null,
+      sortField: 'index',
+      sortAscending: true,
+      scoreProperty: 'score',
+      includeAllSeats: false,
+      showTotals: true,
+      seats: null,
+      showAllRounds: false,
+      roundLabel: 'Round',
+      currentRound: null
+    });
+  }
+
+  applyDeltaToDOM(delta) {
+    if(delta) // Don't call super unless this is a real delta to the scoreboard rather than from a seat
+      super.applyDeltaToDOM(delta);
+    const seats = this.getIncludedSeats();
+    if(seats !== null)
+      this.tableCreate(seats)
+  }
+
+  get(property) {
+    if(property != '_totals')
+      return super.get(property)
+    else if(!this.get('showTotals'))
+      return [];
+    else {
+      // First get total score for each relevant seat
+      const totals = [];
+      let seats = this.getIncludedSeats();
+      if(Array.isArray(seats)) {// Getting seat totals
+        for (const seat of seats) {
+          const score = seat.get(this.get('scoreProperty'));
+          const index = seat.get('index');
+          totals[index] = this.getTotal(asArray(score).map(s => parseFloat(s)));
```

Comment:
This breaks if `index` isn't numeric. I'm not sure if we care.

Reply:
Clearly `index` can be changed to anything. But if we allow for this possibility, we have to return an object, which is perhaps more difficult for the designer to digest in JSON. Personally I'd just document in the wiki section for `seat` that `index` is always expected to be an integer.

Reply:
Fine with me. "If you want stuff to work as expected, make it a number." :rofl:

Reply:
And `TURN` won't work unless `index` is a number, so we already have that expectation in the code.  I will update the wiki to say that `index` should be a number for things to work as expected.

Reply:
OK as is.